client/landing/stepper/declarative-flow/internals/components/step-route/hooks/use-step-route-tracking/index.tsx

```diff
@@ -84,6 +102,8 @@ export const useStepRouteTracking = ( {
 		const pageTitle = `Setup > ${ flowName } > ${ stepSlug }`;
 		recordPageView( pathname, pageTitle );
 
+		return () => handleRecordStepComplete( { flowName, stepSlug, intent } );
```

Comment:
I think I need to rework this. This will run again on dependencies changing, so it probably needs same conditioning with calling `recordStepStart` above.

Reply:
Ok - so the code now (up until [2a720d7](https://github.com/Automattic/wp-calypso/pull/94528/commits/2a720d79d261389b6240b3d120af584e837915dd))  _pressumably_ fixes the issue. In theory only 😬  

It does what I intended it to i.e. to record the step-complete event when (1) the step unmounts and (2) step-start was recorded previously. However, it doesn't work as expected. React Router does things with the routes that won't completely unmount (or won't unmount them immediately). I may have just gone off the wrong intuition here.

I need to rework this again. I think the proper way to target the step may be through the actual render component, not the Route. So may end up moving the whole lot `useStepRouteTracking` around `StepComponent` wrapper instead, which should give me the exit closure / unmount happening as expected.

cc @renancarvalho / @escapemanuele

Reply:
Alternatively, `calypso_signup_actions_step_complete` could fire off next to `calypso_signup_actions_step_submit` (assuming that's something that always fires)

So this may have been another wrong intuition from my end e.g. that it should be controlled at the step/route tracking level. It just seems reasonable to compile it next to the step-start event and on the unmount (as done in Classic) 🤔

Reply:
Looks like my hurdles here were due to a missing key [here](https://github.com/Automattic/wp-calypso/pull/94528/files#diff-1a5cb229a76dfccdc8b471efedc2ab09f2385c9ba00ae3211ba65452ba1c7680R212). 

Now, the elements passed to Route get properly unmounted when switching routes, and the cleanup logic works as expected. I'm not sure if it was intentional not to pass a key though (it could be performance-related). 🤔 

@renancarvalho / @escapemanuele please review this one. It's ready - should log the `calypso_signup_actions_step_complete` upon switching/navigating steps. Consider my comment though and if any ramifications with adding `key` on the elements.